no_std support is broken

I tried using this crate in a no_std context and it doesn't work because the inclusion of `num_traits::Float` isn't behind a feature flag, and evidently, the stuff behind the `libm_works` feature flag doesn't "work".

* `thumbv6m-none-eabi`
* `default-features=false`, `named`
* 
```
error[E0432]: unresolved import `num_traits::Float`
  --> /home/paul/.cargo/git/checkouts/palette-3cb107dfe0152157/1342cce/palette/src/float.rs:12:9
   |
12 | pub use num_traits::Float;
   |         ^^^^^^^^^^^^^^^^^ no `Float` in the root. Did you mean to use `float`?

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: Could not compile `palette`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/palette/src/blend/blend.rs
+++ b/palette/src/blend/blend.rs
@@ -1,6 +1,6 @@
 use num_traits::{One, Zero};
 use float::Float;
-#[cfg(feature = "libm_works")]
+#[cfg(not(feature = "std"))]
 use num_traits::float::FloatCore;
 
 use {cast, clamp, ComponentWise};

--- a/palette/src/blend/equations.rs
+++ b/palette/src/blend/equations.rs
@@ -1,5 +1,5 @@
 use float::Float;
-#[cfg(feature = "libm_works")]
+#[cfg(not(feature = "std"))]
 use num_traits::float::FloatCore;
 
 use {Blend, ComponentWise};

--- a/palette/src/convert.rs
+++ b/palette/src/convert.rs
@@ -391,7 +391,7 @@ where
 ///         0.7353569830524495,
 ///         0.5370987304831942,
 ///     ];
-///     let hsv = Bgr::from_raw_slice(&buffer)[1].into();
+///     let hsv: Hsv<palette::encoding::Srgb, _> = Bgr::from_raw_slice(&buffer)[1].into();
 ///
 ///     assert_relative_eq!(hsv, Hsv::new(90.0, 1.0, 0.5));
 /// }
@@ -441,7 +441,7 @@ where
 ///         blue: 255,
 ///         alpha: 0.3,
 ///     };
-///     let color = css_color.into();
+///     let color: LinSrgba = css_color.into();
 ///
 ///     assert_relative_eq!(color, LinSrgba::new(0.496933, 0.0, 1.0, 0.3));
 /// }

--- a/palette/src/convert.rs
+++ b/palette/src/convert.rs
@@ -352,9 +352,11 @@ where
 ///
 /// use palette::{Component, Hsv, IntoColor, Pixel, Srgb};
 /// use palette::rgb::{Rgb, RgbSpace};
-/// use palette::encoding::Linear;
+/// use palette::encoding::{Linear, self};
 /// use palette::white_point::D65;
 /// use palette::float::Float;
+///
+/// type Hsv64 = Hsv<encoding::Srgb, f64>;
 ///
 /// /// sRGB, but with a reversed memory layout.
 /// #[derive(Copy, Clone, IntoColor, Pixel)]
@@ -391,7 +393,7 @@ where
 ///         0.7353569830524495,
 ///         0.5370987304831942,
 ///     ];
-///     let hsv: Hsv<palette::encoding::Srgb, _> = Bgr::from_raw_slice(&buffer)[1].into();
+///     let hsv: Hsv64 = Bgr::from_raw_slice(&buffer)[1].into();
 ///
 ///     assert_relative_eq!(hsv, Hsv::new(90.0, 1.0, 0.5));
 /// }

--- a/palette/src/float.rs
+++ b/palette/src/float.rs
@@ -9,12 +9,13 @@
 //!
 //! [`libm`]: https://github.com/japaric/libm
 
+#[cfg(feature = "std")]
 pub use num_traits::Float;
 
-#[cfg(feature = "libm_works")]
+#[cfg(not(feature = "std"))]
 pub use self::no_std_float_trait::Float;
 
-#[cfg(feature = "libm_works")]
+#[cfg(not(feature = "std"))]
 mod no_std_float_trait {
     extern crate libm;
     use self::libm::{F32Ext, F64Ext};

--- a/scripts/test_features.sh
+++ b/scripts/test_features.sh
@@ -36,10 +36,10 @@ echo -e "features: $features\n"
 
 #Test without any optional feature
 echo testing with --no-default-features --features "$required_features"
-cargo test --no-default-features --features "$required_features"
+cargo test --release --no-default-features --features "$required_features"
 
 #Isolated test of each optional feature
 for feature in $features; do
 	echo testing with --no-default-features --features "\"$feature $required_features\""
-	cargo test --no-default-features --features "$feature $required_features"
+	cargo test --release --no-default-features --features "$feature $required_features"
 done

